File: my-solutions/0300-longest-increasing-subsequence/solution.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        dp = [1] * len(nums)
        
        for i in range(len(nums)):
            maximum = -1
            for j in range(i-1, -1, -1):
                if nums[i] > nums[j]:
                    maximum = 1 + dp[j] if 1 + dp[j] > maximum else maximum
            dp[i] = maximum if maximum != -1 else dp[i]
        
        return max(dp)

File: my-solutions/0300-longest-increasing-subsequence/solution.cpp
class Solution {
public:
    
    int lengthOfLIS(vector<int>& nums) {
        vector<int> cache(size(nums), 1);
        for(int i = 1; i < size(nums); i++) {
            int max = 1;
            int temp = 0;
            for(int j = 0; j < i; j++) {
                temp = nums[i] > nums[j] ? 1 + cache[j] : 0;
                max = std::max(max, temp);
            }
            cache[i] = max;
        }
        // for(const auto& i: cache) {
        //     cout << i << " ";
        // }
        return *std::max_element(cache.begin(), cache.end());
    }
};
